### ISSUE TITLE
Test our GameItem2d class and ensure that it connects properly to BBDPolygon.

### ISSUE THREAD

**Issue by danbuis** [danbuis/BBDGamesLib#69]
In order to provide access to polygon based methods we have created in the Geometry2d package the 2d game component class has a weak reference to a Polygon object.  Weak as in it doesn't really do much with it unless at construction time the interacts flag is set to true.  If that flag is true then the world coordinates of the GameItem need to stay in lockstep with the BBDPolygon cords.

Should be handled as long as the following methods in both classes track together.

 - [x] translate
 - [x] rotate
 - [x] rotate around point
 - [x] scale
 - [x] scale from point
 - [x] set position
 - [x] set roatation
 - [x] set scale

**Comment by danbuis** [comment]
This will be hard to test as we don't have direct access to the actual vertices of the mesh.  The data is stored 2 deep for us to access with LWJGL, so the best we can do is verify the position matrix object updates correctly.

We can test translate, rotate around point, scale from point, set position to a decent level of confidence.  

The rest will probably be figured out pretty quick is individual functions get used in "production"

**Comment by danbuis** [comment]
Had a fun issue with this one.  So the GameItem2d extends GameItem.  In order for the 2d item to manipulate its position etc it had a local position.  The constructor called the super() constructor, which also set a position.  

Since only the default item had a getPosition() it ended up using that one when I did getPosition(), but anything I did setPosition() which was defined in both since they had different logic, it would update the 2d instance of position.

SOlution was to drop the 2d versions of postion etc and use calls to the superclass
